.github/workflows/_move_ide_tests.yml

```diff
@@ -0,0 +1,127 @@
+          ref: ${{ github.event.inputs.iota_repo_ref || github.ref }}
+
+      - name: Setup Node
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # pin@v4.0.2
+        with:
+          node-version: "20"
+
+      - name: Install dependencies and start Xvfb (emulate display)
+        working-directory: ./external-crates/move/crates/move-analyzer/editors/code
+        run: |
+          sudo apt install libgtk-3-0 -y
+          sudo apt-get install -y xvfb x11-apps x11-xkb-utils libx11-6 libx11-xcb1
+          set -eux
+          # Start server
+          /usr/bin/Xvfb :99 -screen 0 1024x768x24 &
+          sleep 1
+          ps aux | grep Xvfb --color=always | grep -v grep
+          npm install && npm install --save-dev @types/node @types/semver
+          sudo add-apt-repository ppa:kisak/kisak-mesa -y
+          sudo apt update
+          sudo apt upgrade -y
+
+      - name: Build move-analyzer
+        run: |
+          cargo build --bin move-analyzer
+          mkdir -p ~/.iota/bin
+          cp ./target/debug/move-analyzer* ~/.iota/bin
+
+      - name: Run npm test
+        working-directory: ./external-crates/move/crates/move-analyzer/editors/code
+        shell: bash
+        run: DISPLAY=:99.0 npm run test
+
+  move-ide-build:
+    name: Move IDE build
+    runs-on: [self-hosted]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # Pin v4.1.1
+        with:
+          ref: ${{ github.event.inputs.iota_repo_ref || github.ref }}
+
+      - name: Setup Node
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # pin@v4.0.2
+        with:
+          node-version: "20"
+
+      - name: Install dependencies
+        working-directory: ./external-crates/move/crates/move-analyzer/editors/code
+        run: npm install && npm install --save-dev @types/node @types/semver
+
+      - name: Build VSCode extension
+        working-directory: ./external-crates/move/crates/move-analyzer/editors/code
+        run: npm run package
+
+  move-ide-build-with-binaries:
+    name: Move IDE build with released binaries
+    runs-on: [self-hosted]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # Pin v4.1.1
+        with:
+          ref: ${{ github.event.inputs.iota_repo_ref || github.ref }}
+
+      - name: Setup Node
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # pin@v4.0.2
+        with:
+          node-version: "20"
+
+      - name: Install dependencies
+        working-directory: ./external-crates/move/crates/move-analyzer/editors/code
+        run: |
+          npm install
+          npm install --save-dev @types/node @types/semver
+          npm install -g vsce
+          sudo apt install zsh -y
+
+      - name: Build VSCode extensions
+        working-directory: ./external-crates/move/crates/move-analyzer/editors/code
+        run: ./scripts/create.sh -pkg
```

Comment:
The version that is built by the script is hardcoded inside, should we add a parameter to the script here to build the current version?

Reply:
I would say this depends on how we want to do the actual release of the extension, it should probably do the same here to test if this would work? Or we remove this step completely here, I added it mainly to see if we can run this script in the CI to use it potentially for an actual release

Reply:
So, the question is, do we want to integrate the script into the release job? And if we should do it in this PR?

Reply:
Should be done in another PR, created an issue https://github.com/iotaledger/iota/issues/4288